server/lib/deploy/index.js

```diff
@@ -89,12 +89,16 @@ Deploy.prototype._download = function(version, callback) {
     return;
   }
   CURRENT_DOWNLOADS[version] = true;
+  etcdClient = new etcd.Client(self.options.etcd_host, self.options.etcd_port);
+  etcdClient.setSync('/deploys/downloads', JSON.stringify({"version": version, "status": "Downloading"}), {ttl: 3600});
```

Comment:
@rphillips 
Approach: set version and download status and let it live for an hour , as soon as download is complete set status to downloaded for that version.